shared-libs/phone-number/test/libphonenumber.spec.js

```diff
@@ -22,210 +22,237 @@ const settingsWithNoPhoneValidation = {
+    it('returns false for invalid number - two extra letters - trailing', () => {
+      // Insert letters within valid number : they shouldn't be ignored.
+      const actual = phonenumber.validate(settings, NZ_DOMESTIC_VALID + 'ff');
+      assert.isFalse(actual);
+    });
+
+    it('returns false for invalid number - two extra letters - inside', () => {
+      // Insert letters within valid number : they shouldn't be ignored.
+      const actual = phonenumber.validate(settings, '02755526ff36');
+      assert.isFalse(actual);
+    });
+
+    it('returns false for number without country code', () => {
+      const actual = phonenumber.validate({}, NZ_DOMESTIC_VALID);
+      assert.isFalse(actual);
+    });
+
+    it('returns true for number with default country code', () => {
+      const actual = phonenumber.validate(settings, NZ_DOMESTIC_VALID);
+      assert.isTrue(actual);
+    });
+
+    it('returns true for number with explicit country code', () => {
+      const actual = phonenumber.validate({}, NZ_INTERNATIONAL_VALID);
+      assert.isTrue(actual);
+    });
+
+    it('returns true when spaces in number', () => {
+      const actual = phonenumber.validate(settings, '02 75 55 26 36');
+      assert.isTrue(actual);
+    });
+
+    it('returns true when spaces, brackets, dots, dashes in number', () => {
+      // Including space after '+' and unmatched brackets.
+      const actual = phonenumber.validate({default_country_code: '1'}, '+   1 --(80.0 ()()55..6 -8725');
+      assert.isTrue(actual);
+    });
+
+    it('returns false when funky punctuation in number', () => {
+      const actual = phonenumber.validate({default_country_code: '1'}, '+1 <800> 556 8725');
+      assert.isFalse(actual);
+    });
+
+    it('returns true when extra zeros in international format', () => {
+      // Note : that's for NZ normalize. Would be nice to test specific normalizeting issues for target countries.
+      const actual = phonenumber.validate(settings, '+640275552636'); // Unnecessary leading 0 from domestic normalize
+      assert.isTrue(actual);
+    });
+
+    // Test for overly permissive validation
+    // https://github.com/medic/medic/issues/2196
+    it('returns false when given LGs bad phone number example', () => {
+      const actual = phonenumber.validate({default_country_code: COUNTRY_CODES.uganda}, '+25601234');
+      assert.isFalse(actual);
+    });
+
+  });
+
+  describe('same', () => {
+
+    it('returns true for different formats', () => {
+      const actual = phonenumber.same('+640275552636', '64-27-555-2636');
+      assert.isTrue(actual);
+    });
+
+    it('returns false for different numbers', () => {
+      const actual = phonenumber.same('+640275552637', '64-27-555-2636');
+      assert.isFalse(actual);
+    });
+
+    it('missing country code matches default', () => {
+      const actual = phonenumber.same('+41446681800', '446681800');
+      assert.isTrue(actual);
+    });
```

Comment:
I added this section to bring the unit tests closer to the code. It's basically a copy of the transition messages test that was failing before.